torch/fx/experimental/symbolic_shapes.py

```diff
@@ -1253,27 +1278,35 @@ def _get_key(self):
         """
         return (len(self.replacements), len(self.divisible))
 
-    def _produce_dyn_sizes(self, ex: torch.Tensor, source: Source) -> List[sympy.Expr]:
+    def _produce_dyn_sizes(self,
+                           ex: torch.Tensor,
+                           source: Source,
+                           dims: Optional[List[DIM_DYNAMISM_STATE]]) -> List[sympy.Expr]:
```

Comment:
what does it mean when this is None

Reply:
It means we just assume DUCK for all, the back compat state. Hinted at on the `DIM_DYNAMISM_STATE` enum, but probably could be spelled out clearer here. Would you like a comment to that effect?